File: 49-group-anagrams/49-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        mp = collections.defaultdict(list)
        
        for str in strs:
            key = [0] * 26
            for ch in str:
                key[ord(ch) - ord('a')] += 1
            mp[tuple(key)].append(str)
            
        return [group for _, group in mp.items()]

File: 49-group-anagrams/49-group-anagrams.cpp
#include <vector>
#include <string>
#include <array>
#include <map>

using namespace std;

class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        map<array<int, 26>, vector<string>> anagram_grp;
        for (string str : strs) {
            array<int, 26> key{0};
            for (char ch : str)
                key[ch - 'a'] += 1;
            anagram_grp[key].push_back(str);
        }
        
        vector<vector<string>> res;
        
        for (auto pair : anagram_grp)
            res.push_back(pair.second);
        return res;
    }
};